### ISSUE TITLE
abilities design discussion

### ISSUE THREAD

**Issue by Xerandael** [Xerandael/medieval-future#2]
interface:
```
() => movements().map({x, y} => { ... }
```

**Comment by Xerandael** [comment]
These are some ideas I have in rough-draft form.
 - deflect:  with a melee weapon, strike at another unit's weapon, lowering their defense and aim till next turn
 - suppression:  with a ranged weapon, lower another unit's aim

**Comment by Xerandael** [comment]
I'm assigning myself for now just to put out a couple early mechanics, but I'll leave this open and unassign myself.

**Comment by Xerandael** [comment]
TODO:  work out composable geometry for generating interesting targetting mechanisms.  Subtractive or additive synthesis?  Basic shapes plus iterators within those shapes.

TODO:  work out interesting states for units.  Maybe some kind of one-two combos where just dealing damage isn't enough against some units, but there has to be a specific combination of status applications to bring them down.  The goal is for combat to be an interesting logical puzzle.

In short, tactical decisions should be part geometry problem, part ordering problem, and part boolean satisfiability problem.
